spec/unit/BridgedClient.spec.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+const { BridgedClient, BridgedClientStatus } = require("../../lib/irc/BridgedClient.js");
+
+const STATE_DISC = {
+    status: BridgedClientStatus.DISCONNECTED
+}
+
+const STATE_CONN = {
+    status: BridgedClientStatus.CONNECTED,
+    client: {}
+}
+
+const STATE_CONN_MAX5 = {
+    status: BridgedClientStatus.CONNECTED,
+    client: {
+        supported: {
+            nicklength: 5
+        }
+    }
+}
+
+describe("BridgedClient", function() {
+    describe("getValidNick", function() {
+        it("valid nick unchanged", function() {
+            expect(BridgedClient.getValidNick("foobar", true, STATE_DISC)).toBe("foobar");
```

Comment:
it would be good to have more valid examples here containing numbers and special characters